src/dicarlo_lab_to_nwb/conversion/stimuli_interface.py

```diff
@@ -120,24 +120,51 @@ def add_to_nwbfile(
 
         columns = mwkorks_df.columns
         assert ground_truth_time_column in columns, f"Column {ground_truth_time_column} not found in {columns}"
+        image_presentation_time_seconds = mwkorks_df[ground_truth_time_column] / 1e6
         stimulus_ids = mwkorks_df["stimulus_presented"]
+        stimulus_filenames = mwkorks_df["stimulus_filename"].to_list()
+        stimulus_hashes = mwkorks_df["image_hash"].to_list()
 
-        unique_stimulus_ids = stimulus_ids.unique()
-        unique_stimulus_ids.sort()
+        # stimulus size and durations are constant for all images
+        stimulus_on_s = np.unique(mwkorks_df["stim_on_time_ms"]) / 1e3
+        stimulus_off_s = np.unique(mwkorks_df["stim_off_time_ms"]) / 1e3
+        stimulus_size_deg = np.unique(mwkorks_df["stimulus_size_degrees"])
+        # Generate unique lists from the above example while preserving the order of the unique values from the first list (stimulus_indices)
+        seen = set() # Initialize an empty set to track seen values from list1
+
+        # Initialize empty lists to store the unique values
+        uniq_stim_indices = []
+        uniq_stim_filenames = []
+        uniq_stim_hashes = []
+
+        # Iterate through the lists and preserve unique values based on list1
+        for l1, l2, l3 in zip(stimulus_ids, stimulus_filenames, stimulus_hashes):
+            if l1 not in seen:
+                uniq_stim_indices.append(l1)
+                uniq_stim_filenames.append(l2)
+                uniq_stim_hashes.append(l3)
+                seen.add(l1)
+
+        # Sort the paired lists based on the first list (stimulus_ids)
+        paired_uniq_lists = list(zip(uniq_stim_indices, uniq_stim_filenames, uniq_stim_hashes))
+        paired_uniq_lists.sort()
+        uniq_stim_indices, uniq_stim_filenames, uniq_stim_hashes = zip(*paired_uniq_lists)
+        assert len(uniq_stim_indices) == len(uniq_stim_filenames), "Stimulus indices and filenames do not match"
 
         image_list = []
-        image_mode_to_nwb_class = {"L": GrayscaleImage, "RGB": RGBImage, "RGBA": RGBAImage}
+        image_mode_to_nwb_class = {"LA": GrayscaleImage, "L": GrayscaleImage, "RGB": RGBImage, "RGBA": RGBAImage}
```

Comment:
Interesting, so this was failng : O
GrayScale with luminance as you were mentioning.

Reply:
Yeah, this is the reason I am using the extended class and added an if-statement with the following: https://github.com/catalystneuro/dicarlo-lab-to-nwb/blob/0aa9a6edc24b83b6fb674a8065afc37003f8291c/src/dicarlo_lab_to_nwb/conversion/stimuli_interface.py#L244-L246